### PR TITLE
doc: net: Move networking chapters around

### PR DESCRIPTION
**Pull request by jukkar**
Trying to group the networking chapters so that the layout is
more logical.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Comment by jukkar** [comment]
The networking doc layout would look something like this with this PR.
zephyr-networking-chapters

File: doc/reference/networking/apis.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+   ip_4_6.rst
+   dns_resolve.rst
+   net_hostname.rst
+   net_mgmt.rst
+   net_offload.rst
+   net_timeout.rst
+   net_stats.rst
+   net_context.rst
+   promiscuous.rst
+   trickle.rst
```
Comment (rveerama1) [inline doc/reference/networking/apis.rst]: Overall looks good. This section is still mixed with different things.

**Comment by rlubos** [comment]
I like it, looks pretty neat. Do we want to reference Thread (OpenThread) under `Network Technologies`? I can prepare a page for that.

**Comment by nashif** [comment]
looks good, way to go.

**Comment by jukkar** [comment]
> I like it, looks pretty neat. Do we want to reference Thread (OpenThread) under `Network Technologies`? I can prepare a page for that.

That sounds like a good plan so please do.

I will submit a proper PR for this after the document changes that I submitted this week have been merged.

**Comment by dbkinder** [comment]
I like too!  BTW, you can see the Shippable doc build artifacts for any PR: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/13616/docs/index.html    (adjust the PR number in the URL)

**Comment by jukkar** [comment]
Updated the chapter location a bit. I also noticed that there are still couple of networking reference files that are missing Overview chapter. I will send new PRs for these.

**Comment by jukkar** [comment]
> BTW, you can see the Shippable doc build artifacts for any PR: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/13616/docs/index.html (adjust the PR number in the URL)

Cool, I did not know this. Thanks for the info!

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
thanks!

File: doc/reference/networking/system_mgmt.rst [hunk 2]
```diff
@@ -0,0 +1,18 @@
+.. _net_system_mgmt:
+
+Network System Management
```
Comment (pfalcon) [inline doc/reference/networking/system_mgmt.rst]: I would add `(TSN)` at the end of title here, as it's commonly known by abbreviation.

**Review by pfalcon — APPROVED** [review]

**Pull request merged**